listener/src/event_handler/auction_settlement_handler.rs

```diff
@@ -0,0 +1,211 @@
+    }
+}
+
+impl Debug for AuctionSettlementHandler {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "AuctionSettlementHandler")
+    }
+}
+
+
+impl EventHandler for AuctionSettlementHandler {
+    fn process_event(
+        &self,
+        logger: Logger,
+        _block: Arc<EthereumBlock>,
+        _transaction: Arc<Transaction>,
+        log: Arc<Log>,
+    ) -> Result<Vec<EntityOperation>, Error> {
+        let mut event_data: Vec<u8> = log.data.0.clone();
+        info!(logger, "Parsing Auction Settlement event from {} bytes: {:?}", event_data.len(), event_data);
+        let auction_id = U256::pop_from_log_data(&mut event_data);
+        let state_index = U256::pop_from_log_data(&mut event_data).saturating_sub(U256::one());
+        let new_state_hash = H256::pop_from_log_data(&mut event_data);
+
+        // Strange info coming with packed bytes
+        let _bytes_init = u16::pop_from_log_data(&mut event_data);
+        let _byte_size = u16::pop_from_log_data(&mut event_data);
+
+        // Remaining information is the Auction Results
+        let encoded_solution = event_data;
+
+        info!(logger, "Parsing packed Auction Results from {} bytes: {:?}", encoded_solution.len(), encoded_solution);
+        let auction_results = Solution::from_bytes(encoded_solution);
+        debug!(logger, "Parsed Auction Results: {:?}", auction_results);
+
+        // Fetch relevant information for transition (accounts, orders, parsed solution)
+        let mut account_state = self.store
+            .get_balances_for_state_index(&state_index)
+            .map_err(|e| failure::err_msg(format!("{}", e)))?;
+
+        let mut orders = self.store
+            .get_orders_of_slot(&auction_id)
+            .map_err(|e| failure::err_msg(format!("{}", e)))?;
+
+        let standing_orders = self.store
+            .get_standing_orders_of_slot(&auction_id)
+            .map_err(|e| failure::err_msg(format!("{}", e)))?;
+
+        orders.extend(standing_orders
+            .iter()
+            .filter(|standing_order| standing_order.num_orders() > 0)
+            .flat_map(|standing_order| standing_order.get_orders().clone())
+        );
+        info!(logger, "Found {} valid Orders for this auction", orders.len());
+        account_state.apply_auction(&orders, auction_results);
+
+        let mut entity: Entity = account_state.into();
+
+        // Set the state root according to event information
+        entity.set("id", new_state_hash.to_value());
+        Ok(vec![
+            EntityOperation::Set {
+                key: util::entity_key("AccountState", &entity),
+                data: entity,
+            }
+        ])
+    }
+}
+
+#[cfg(test)]
+pub mod unit_test {
+    use super::*;
+    use dfusion_core::database::tests::DbInterfaceMock;
+    use dfusion_core::models::{AccountState, TOKENS, Order, BatchInformation, StandingOrder};
+    use web3::types::{Bytes, H256, U256};
+
+    #[test]
+    fn test_from_log() {
```

Comment:
Can we add tests for the unhappy cases as well?

Reply:
Done!